src/stk/molecular/topology_graphs/host_guest/complex.py

```diff
@@ -133,36 +186,98 @@ def __init__(
             If `guest_start` or `guest_target` is defined but the other
             is not.
 
+        :class:`ValueError`
+            If the old API and new API are used simultaneously.
+
         """
 
-        num_nones = sum(
-            1 for vector in (guest_start, guest_target)
-            if vector is None
-        )
-        if num_nones == 1:
-            raise TypeError(
-                'If guest_start or guest_target is defined, '
-                'the other must be too.'
+        if guests is not None and guest is not None:
+            raise ValueError(
+                'You are attempting to use the old API with the new '
+                'API. Either use the stk.host_guest.Guest API and '
+                '`guests`, or the old API.'
             )
 
-        if guest_start is None:
-            start = target = (1., 0., 0.)
-        else:
-            start = guest_start = tuple(guest_start)
-            target = guest_target = tuple(guest_target)
+        if guests is not None and displacement is not None:
```

Comment:
These ifs can be rolled into one
```python
if guests is not None and None in {guest, displacement, guest_start, guest_target}
```

Reply:
I do not think this does what you want?

Wrote an equivalent check.